membrane_curvature/tests/test_membrane_curvature.py

```diff
@@ -206,7 +207,38 @@ def test_get_z_surface(x_bin, y_bin, x_range, y_range, expected_surface):
                             [0., 0., 150.], [100., 100., 150.], [200., 100., 150.],
                             [0., 200., 150.], [100., 200., 150.], [200., 200., 150.]])
     surface = get_z_surface(dummy_array, x_bin, y_bin, x_range, y_range)
-    assert_almost_equal(surface, expected_surface)
+    assert_allclose(surface, expected_surface)
+
+
+@pytest.mark.parametrize('dummy_surface, expected_interpolated_surface', [
+    # array 3x3 with all 150 and one nan
+    (np.array(([150., 150., 150.],
+               [150., np.nan, 150.],
+               [150., 150., 150.])),
+     np.full((3, 3), 150.)),
+    # array 3x4 with all 150 and two nans
+    (np.array([[150.,  150, 150.,  150.],
+               [150., np.nan, np.nan,  150.],
+               [150., 150., 150.,  150.]]),
+     np.full((3, 4), 150.)),
+    # array 4x4 with all 150 and two nans
+    (np.array([[150., 150,  150.,  150.],
+               [150., np.nan, np.nan,  150.],
+               [150., 130., 140.,  150.],
+               [150., 150., 150.,  150.]]),
+     np.array([[150., 150, 150.,  150.],
+               [150., 140., 145.,  150.],
+               [150., 130., 140.,  150.],
+               [150., 150., 150.,  150.]])),
+    # array 3x3  with lots of nans
+    (np.array([[np.nan, np.nan, 150.],
+              [150, np.nan, 150.],
+              [np.nan, 150., np.nan]]),
+     np.full((3, 3), 150.))
```

Comment:
Could you please add some tests with more interesting values? e.g. in the one below, the 145 number could have come from interpolating either along the x-axis or the y-axis. In order to be diagnostic of potential current and future bugs, I'd be interested in seeing one where using 1D interpolation gives different results if it's run across the x-axis, vs. the y-axis.

```python
(np.array([[150., 150,  150.,  150.],
               [150., np.nan, np.nan,  150.],
               [150., 130., 140.,  150.],
               [150., 150., 150.,  150.]]),
     np.array([[150., 150, 150.,  150.],
               [150., 140., 145.,  150.],
               [150., 130., 140.,  150.],
               [150., 150., 150.,  150.]])),
```